diff -rup keepassx-0.4.3.orig/src/lib/random.cpp keepassx-0.4.3/src/lib/random.cpp
--- keepassx-0.4.3.orig/src/lib/random.cpp	2015-08-09 03:22:31.425433486 -0700
+++ keepassx-0.4.3/src/lib/random.cpp	2015-08-09 03:24:21.408270493 -0700
@@ -29,6 +29,10 @@
 	#include <QSysInfo>
 #endif
 
+#ifndef Q_WS_WIN
+    #include <unistd.h>
+#endif
+
 #include <QCryptographicHash>
 #include <QCursor>
 #include <QDataStream>
