diff -rup smplayer.orig/src/prefinput.cpp smplayer/src/prefinput.cpp
--- smplayer.orig/src/prefinput.cpp	2016-02-02 03:12:54.683839378 -0800
+++ smplayer/src/prefinput.cpp	2016-02-02 03:13:41.053080198 -0800
@@ -56,6 +56,8 @@ void PrefInput::createMouseCombos() {
 	left_click_combo->addItem( tr("Pause"), "pause" );
 	left_click_combo->addItem( tr("Pause / Frame step"), "pause_and_frame_step" );
 	left_click_combo->addItem( tr("Stop"), "stop" );
+	left_click_combo->addItem( tr("Play previous"), "play_prev" );
+	left_click_combo->addItem( tr("Play next"), "play_next" );
 	left_click_combo->addItem( tr("Go backward (short)"), "rewind1" );
 	left_click_combo->addItem( tr("Go backward (medium)"), "rewind2" );
 	left_click_combo->addItem( tr("Go backward (long)"), "rewind3" );
